### PR TITLE
Switch to GitHub actions

### LINKED ISSUE #99: setup github action
Remove travis job in favour of github action.
Look at this great template for how to do it, done by @jmrtt
https://github.com/awalterschulze/gographviz/pull/66/files

### PR DESCRIPTION
**Pull request by cornelk**
Closes #99

### CODE REVIEW

File: internal/test/t1/Makefile [hunk 1]
```diff
@@ -1,2 +1,2 @@
 regenerate:
-	gocc t1.bnf
\ No newline at end of file
+	gocc -p github.com/goccmack/gocc/internal/test/t1 t1.bnf
```
Comment (awalterschulze) [inline internal/test/t1/Makefile:2]: All looks great to me, is the `-p`s really necessary or can we set the GOPATH as part of the CI setup?

**Review by awalterschulze — APPROVED** [review]
Really great job.
Thank you so much for setting this up.
We really appreciate it.

**Comment by awalterschulze** [comment]
I am going to merge this, so github actions start running and it can be iterated on, if there are any issues.

**Comment by awalterschulze** [comment]
@cornelk  one more thing is to swap the picture in the Readme that links to travis to rather link to github actions.

File: internal/test/t1/Makefile (hunk 1 above)
Comment (cornelk) [reply internal/test/t1/Makefile:2]: I had multiple issues where gocc did put parts of my the user home path as include path if I did not force it by using -p, so I put it here by default.  You can see an example in this failed CI run: https://github.com/cornelk/gocc/runs/1793013769

```
git diff .
diff --git a/example/astx/errors/errors.go b/example/astx/errors/errors.go
index 640214b..666ce02 100755
--- a/example/astx/errors/errors.go
+++ b/example/astx/errors/errors.go
@@ -6,7 +6,7 @@ import (
 	"fmt"
 	"strings"
 
-	"github.com/goccmack/gocc/example/astx/token"
+	"home/runner/work/gocc/gocc/example/astx/token"
 )

```

Comment (awalterschulze) [reply internal/test/t1/Makefile:2]: Are you sure setting the GOPATH does not help?

Comment (awalterschulze) [reply internal/test/t1/Makefile:2]: This is the logic I think of how the default out dir is chosen https://github.com/goccmack/gocc/blob/47605ef073d33e0a47a9209d61fa0eff182a95c6/internal/config/config.go#L204

**Pull request merged**